### PR TITLE
Week5 Assignment

### PR DESCRIPTION
**Pull request by janewaitara**

### CODE REVIEW

File: Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/LoginFragment.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                }
+            }
+        }
+    }
+
+    /**
+     * a fun to validate whether the login password is empty or shorter than expected.
+     * Returns a Boolean
+     * */
+    private fun validatePassword(): Boolean{
```
Comment (jenndotcodes) [inline Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/LoginFragment.kt:42]: Nice job on the validation here.

File: Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/LoginFragment.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                true
+            }
+        }
+    }
+
+    /**
+     * a fun to validate whether the login userName is empty or shorter than expected.
+     * Returns a Boolean
+     * */
+    private fun validateUserName(): Boolean{
```
Comment (jenndotcodes) [inline Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/LoginFragment.kt:65]: Nice job on the validation here as well.

File: Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/MovieAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class MovieAdapter(val clickListener: MovieListClickListener) : RecyclerView.Adapter<MovieAdapter.MovieViewHolder>() {
+
+    private var movieList = mutableListOf(
```
Comment (jenndotcodes) [inline Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/MovieAdapter.kt:12]: You've done a nice job setting up your dao and database but unfortunately the recyclerview is still displaying this list and not storing it in the database.

File: Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/MovieSharedPrefs.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.janewaitara.movieapp
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class MovieSharedPrefs {
```
Comment (jenndotcodes) [inline Week 5/MovieAppW5/app/src/main/java/com/janewaitara/movieapp/MovieSharedPrefs.kt:6]: Nice job on the shared prefs, this functionality works well in the app.

**Review by jenndotcodes — CHANGES_REQUESTED** [review]
Hi Jane, nice work on this app and nice job on the login screen and the shared preferences, and you've included a logout button. Overall, you've used a good style and the code is easy to follow. I would suggest to create some sub packages to organize things as our architecture gets more complicated, and the list is still populated from the hardcoded list of data and not putting the data into the database so I have to assign 'room for improvement', but you did set the database up. It needs to be utilized as well. Nice work overall, and I look forward to your submission next week!

**Comment by janewaitara** [comment]
Hello Jenn,

Should I work on it again for corrections for me to merge or should I
implement the changes on week 6 assignment?

I feel heart broken for the failure😭😢, please if it's allowed share the
leader board for our scores.

Regards,
Jane Waitara.

**Comment by jenndotcodes** [comment]
Hi Jane,

No worries, we don't have a leader board but you're doing quite well and I
wouldn't be discouraged at all, this was a tough assignment. You don't have
to make the changes, you can merge the changes and continue on with next
week. You're in excellent standing and doing well so far, keep going! We
have a lot more concepts to cover and this week looks like it was difficult
all around.

Keep up the good work, I'm excited to see program 6, you're doing very well
so far.

Jenn

**Pull request merged**